src/apple/network.rs

```diff
@@ -237,4 +251,16 @@ impl NetworkExt for NetworkData {
     fn total_errors_on_transmitted(&self) -> u64 {
         self.errors_out
     }
+
+    fn mac_address(&self) -> MacAddr {
+        self.mac_addr
+    }
+
+    fn ipv4_address(&self) -> Ipv4Addr {
+        self.ipv4_addr
```

Comment:
One big question about this: if it's an IPv6 interface, will there be an IPv4 address and netmask? If not, we might maybe want to store both the address and the netmask into an enum looking like this:

```rust
enum IPInfo {
    Unknown,
    IPv4 { address: Ipv4Addr, netmask: Ipv4Addr },
    IPv6 { address: Ipv6Addr, netmask: Ipv6Addr },
}
```

What do you think?

Reply:
Ok. I have written a [c program](https://gist.github.com/kayoch1n/6b68080442ae3d909a5ecd78587cfd76#file-getifaddrs_macos-c) to verify this. On macos, there are VPN interfaces containing IPv6 address only. The c program above outputs something like this

```
utun0: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 1380
	inet6 fe80::751f:xxxx:yyyy:zzzz%utun0 prefixlen 64 scopeid 0x12 
	nd6 options=201<PERFORMNUD,DAD>
utun1: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 2000
	inet6 fe80::214c:xxxx:yyyy:zzzz%utun1 prefixlen 64 scopeid 0x13 
	nd6 options=201<PERFORMNUD,DAD>
```
I think we should make these fields(mac, ipv4 and ipv6) optional. Moreover, [this link](https://networkengineering.stackexchange.com/a/23758) indicates a VLAN interface with **multiple IPv4 addresses** available, though I haven't seen this case before. Things get complicated😮‍💨.  A naive solution for this issue can be implemented by storing Ipv4Addr/Ipv6Addr in different vectors.
```rust
struct NetworkData {
    // ...
    mac_addr: Optional<MacAddr>,
    ipv4_addrs: Vec<(Ipv4Addr, Ipv4Addr)>, 
    // ipv6_addrs for IPv6 respectively
}
```
Upon this solution, when a user intends to obtain the IPv4 address of an interface, he will have to call `Vec::first()` or `Iterator::next()` indirectly, depending on what is declared in traits, to get an `Option`. 
I am not sure whether this solution can cover all cases. If not, I'd prefer to remove IP-related fields and keep the MAC address only. Still waiting for a better solution...